package/src/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -117,13 +168,37 @@ const renderImage = ({
     }
   };
 
+  const onPressVideo = () => {
+    if (selected) {
+      setSelectedFiles((files) => files.filter((file) => file.uri !== asset.uri));
+    } else {
+      setSelectedFiles((files) => {
+        if (files.length >= maxNumberOfFiles) {
+          return files;
+        }
+        return [
+          ...files,
+          {
+            duration,
+            name: asset.filename,
+            size: asset.fileSize,
+            type: 'video',
+            uri: asset.uri,
+          },
+        ];
+      });
+    }
+  };
+
   return (
     <AttachmentImage
       ImageOverlaySelectedComponent={ImageOverlaySelectedComponent}
       numberOfAttachmentPickerImageColumns={numberOfAttachmentPickerImageColumns}
-      onPress={onPress}
+      onPress={isImage === 'image' ? onPressImage : onPressVideo}
```

Comment:
We should maybe rename the component to reflect that it can be either an image or a video? Or better, make one `AttachmentImage` and one `AttachmentVideo` component, that can render a common `AttachmentSomethingOrOther` so the differences are split into the separate components